frame/staking/src/pallet/mod.rs

```diff
@@ -1593,6 +1596,12 @@ pub mod pallet {
 			//   threshold bond required.
 			//
 			// Otherwise, if caller is the same as the controller, this is just like `chill`.
+
+			if Nominators::<T>::contains_key(&stash) && Nominators::<T>::get(&stash).is_none() {
```

Comment:
Seems like this should be a single operation storage API

Reply:
actually, the `contains_key` is pretty pointless, right? I was trying to prevent extra decoding, but it was wrong. If the item exists, then we decode anyways since `contains_key` returns true and we move to decoding. If it does not exist, we return the `default` anyways.

Reply:
ahhhh no I am wrong, this needs to be two different operations, because the `get(_)` is is going to fail decoding and return the default, but the key will exist.

Reply:
happy to talk about this in another issue/PR, but for the purpose of this, I think we're good.

Reply:
What I am saying is that maybe there is a TODO to add a new API which does what you are doing here, but not going to storage twice